### PR TITLE
FEM-1140 ad cue points

### PR DESCRIPTION
**Pull request by gal-orlanczyk**
* Added ad cue points event for VMAP ads.
* Added AdInfo object to AdStarted event.
* Added warm up option for plugins.
* Moved all ad related classes to IMA pod spec instead of Core.

### CODE REVIEW

File: Plugins/IMA/AdEvent.swift [hunk 1]
```diff
@@ -0,0 +1,151 @@
+//
+//  AdEvent.swift
+//  Pods
+//
+//  Created by Gal Orlanczyk on 07/03/2017.
+//
+//
+
+import Foundation
```
Comment (OrenMe) [inline Plugins/IMA/AdEvent.swift]: Why move ad events to IMA?
They should be general and implemented as part of other ad plugins.

File: Classes/Managers/PlayKitManager.swift [hunk 2]
```diff
@@ -48,6 +48,9 @@ import UIKit
     }
     
     @objc public func registerPlugin(_ pluginClass: BasePlugin.Type) {
+        if let pluginWarmUp = pluginClass as? PKPluginWarmUp.Type {
```
Comment (OrenMe) [inline Classes/Managers/PlayKitManager.swift]: why is warmup changes included in this PR? it is titled adCuePoints

File: Plugins/IMA/IMAPlugin.swift [hunk 3]
```diff
@@ -248,6 +233,7 @@ extension PKErrorDomain {
     private func updateAdTag(_ adTag: String, tagTimeKeyForRemove: TimeInterval) {
         self.tagsTimes![tagTimeKeyForRemove] = nil
         self.destroyManager()
+        self.contentComplete()
```
Comment (OrenMe) [inline Plugins/IMA/IMAPlugin.swift]: Is this a bug fix? why always call contentComplete when updating a tag?

Comment (gal-orlanczyk) [inline Plugins/IMA/IMAPlugin.swift]: This is for manual ad tags list only.

When requesting ads from the same URL in a short time you can receive error for ad could not been loaded and this fixes it. 
If you think this should never happen I will remove this.

File: Plugins/IMA/AdEvent.swift (hunk 1 above)
Comment (gal-orlanczyk) [inline Plugins/IMA/AdEvent.swift]: @noamtamim, noam mentioned the same issue and I understand the concern but right now our Ads events are not general and they are more IMA specific In Addition when we will want to separate the plugins we will not be able to post this events as we make them internal.

In my opinion having events as part of the plugin makes much more sense.

Update me how to proceed here.

File: Classes/Managers/PlayKitManager.swift (hunk 2 above)
Comment (gal-orlanczyk) [inline Classes/Managers/PlayKitManager.swift]: I was a small change and it helps improves IMA loading.

File: Plugins/IMA/AdEvent.swift (hunk 1 above)
Comment (OrenMe) [inline Plugins/IMA/AdEvent.swift]: @gal-orlanczyk please revert this, we will refactor ad events if needed.

File: Classes/Managers/PlayKitManager.swift (hunk 2 above)
Comment (OrenMe) [inline Classes/Managers/PlayKitManager.swift]: @gal-orlanczyk the change is necessary - it is best practice to separate PR to specific change for tracking and being able to revert.
let's leave it for now but keep in mind for future PR.

File: Plugins/IMA/IMAPlugin.swift (hunk 3 above)
Comment (OrenMe) [inline Plugins/IMA/IMAPlugin.swift]: This could be, let's flag this for refactor for your suggestion for separate handlers for manual and vmap.

**Pull request merged**